pkg/k8sutil/k8sutil.go

```diff
@@ -183,3 +184,44 @@ func isKubeMetaKind(kind string) bool {
 func isRunModeLocal() bool {
 	return os.Getenv(ForceRunModeEnv) == string(LocalRunMode)
 }
+
+// SupportsOwnerReference func checks whether a given dependent supports owner references, based in the owner.
+func SupportsOwnerReference(restMapper meta.RESTMapper, owner runtime.Object, dependent runtime.Object) (bool, error) {
```

Comment:
Is it a check or a SetOwnerReference. 
Funcs that return boll should has the prefix is or has 
Also, shows that it is missing the cohesion.

Reply:
I think this function name is intuitive as is. `Is` and `Has` are verbs describing booleanness of the suffix. `Supports` is a verb for the booleanness of whether the args support owner references. Any other function name (e.g. `hasSupportForOwnerRef` as you suggest), just makes the name longer with no added clarity or readability, IMO.

Reply:
I do not agree @joelanford.

`SupportsOwnerReference ` lead to me think that is `setting/adding` something since the naming conventions for boolean methods requires that they should start with is or has in order to make intuitive that it will return true or face. 

Also, why an error is returned. Should not it return false if an error is faced? Is the error useful result as well?

Reply:
Hi @joelanford 

I needed to read with attention all func to know that it is checking something which would be intuitive and got in 1 second if we just follow the common practice for naming conventions. PS.: I prefer big names that make clear what the method does then check all impl to get that. 

Note that `SupportsOwnerReference ` lead to think that is `setting/adding/doing` something.

Also, why an error is returned. Should not it return false if an error is faced? Is the error useful result as well?

Reply:
PS.: same for the var.

Reply:
I believe `restMapper.RESTMapping` , these functions are returning `err` as out params, Hence to bubble these errors up, `error` has been made part of signature.
IMO: `Supports` gets an answer of `true` or `false` even gramatically. I believe `Supports` conveys the meaning, and that is the purpose.

Reply:
> SupportsOwnerReference lead to me think that is setting/adding something

Why? The function name doesn't say anything about setting or adding. This is about as plain and simple English as I can imagine this function name being. When `Is` and `Has` make sense to use, I agree that they're prefered. But here, we're not asserting whether the thing `is` something or `has` something.

> I needed to read with attention all func to know that it is checking which would be intuitive and got in 1 second if we just follow the common practice for naming conventions. PS.: I prefer big names that make clear what the method does then check all impl to get that.

The [official go docs](https://golang.org/doc/effective_go.html#package-names) touch on this specifically:
> Another short example is once.Do; once.Do(setup) reads well and would not be improved by writing once.DoOrWaitUntilDone(setup). Long names don't automatically make things more readable. A helpful doc comment can often be more valuable than an extra long name.

As @estroz suggested in another comment, the GoDoc on this function should be a bit more descriptive, and that aligns with the official Go doc suggestion that doc comments can be more valuable than a long name.

---

> Also, why an error is returned. Should not it return false if an error is faced? Is the error useful result as well?

Because checking the rest mapping can result in an error, which means we don't know what is supported. Returning `false` would mean that we would add the owner annotation, and that might not be the correct thing to do.

Reply:
@camilamacedo86 Please let me know if this clarifies your concerns